File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        st = set()
        ans = 0
        start = 0
        for idx, char in enumerate(s):
            while char in st:
                st.remove(s[start])
                start+=1
            ans = max(ans, idx-start+1)    
            st.add(char)
        return ans;    
                
            

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> st;
        int start = 0;
        int ans = 0;
        for(int i=0;i<s.size();i++){
            
            while(st.find(s[i]) != st.end()){
                st.erase(s[start]);
                start++;
            }
            st.insert(s[i]);
            ans = max(ans, i - start + 1 );
        }

        return ans;
    }
};